src/context/AnnouncementModalContext.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+
+export const AnnouncementModalContext = React.createContext([
+  undefined,
+  undefined,
+]);
+
+type AnnouncementModalContext = {
+  visible: boolean;
+  visibleAnnouncementUid: null | string;
+};
+
+export const AnnouncementModalProvider = ({ children }) => {
+  const state = useState<AnnouncementModalContext>({
+    visible: false,
+    visibleAnnouncementUid: null,
+  });
```

Comment:
```suggestion
type AnnouncementModalContext = {
  visible: boolean;
  visibleAnnouncementUid?: string;
};

export const AnnouncementModalProvider = ({ children }) => {
  const state = useState<AnnouncementModalContext>({
    visible: false,
    visibleAnnouncementUid: undefined,
  });
```

I would opt for undefined over null where we can